### ISSUE TITLE
Searching K nearest neighbors for each test sample in training set

### ISSUE THREAD

**Issue by isablle31** [saurfang/spark-knn#40]
Is there a way to extract the K nearest neighbors from Training samples from the KNN model in the Scala version?

**Comment by saurfang** [comment]
Yes. You simply run `transform` again on the training data to get the K neighbors for every point in the dataset. https://github.com/saurfang/spark-knn/blob/master/spark-knn-core/src/test/scala/org/apache/spark/ml/knn/KNNSuite.scala#L31
